rs/bitcoin/validation/src/header.rs

```diff
@@ -219,29 +132,37 @@ fn find_next_difficulty_in_chain(
 ) -> u32 {
     // This is the maximum difficulty target for the network
     let pow_limit_bits = pow_limit_bits(network);
+
     match network {
         Network::Testnet | Network::Regtest => {
             let mut current_header = *prev_header;
             let mut current_height = prev_height;
-            let mut current_hash = prev_header.block_hash();
+            let mut current_hash = current_header.block_hash();
             let initial_header_hash = store.get_initial_hash();
 
             // Keep traversing the blockchain backwards from the recent block to initial
             // header hash.
-            while current_hash != initial_header_hash {
+            loop {
+                // Check if non-limit PoW found or it's time to adjust difficulty.
                 if current_header.bits != pow_limit_bits
                     || current_height % DIFFICULTY_ADJUSTMENT_INTERVAL == 0
                 {
                     return current_header.bits;
                 }
 
-                // Traverse to the previous header
-                let header_info = store
-                    .get_header(&current_header.prev_blockhash)
+                // Stop if we reach the initial header.
+                if current_hash == initial_header_hash {
```

Comment:
nit: wasn't it more readable with the while loop ?

Reply:
True